src/tidylinreg/tidylinreg.py

```diff
@@ -0,0 +1,199 @@
+class LinearModel:
+    '''
+    A Linear Model class for various regression tasks, implemented in the style of the R lm()
+    function and able to perform bootstrapped estimations.
+    '''   
+    
+    def __init__(self, method: str = 'parametric'):
+        '''
+        Initialize LinearModel class.
+        
+        Parameters
+        ----------
+        method : str, optional
+            One of ('parametric','bootstrap'). If 'bootstrapped', parameter estimates, standard errors, etc. will be computed using bootstrapping
+            from the fitted data. if 'parametric', classical parametric estimates are computed.
+        '''
+        self.__method = method
+        self.params = None
+        self.param_names = None
+        self.X = None
+        self.y = None
+  
+          
+    def fit(self,y,X):
```

Comment:
Changing the function input to take X followed by y, as per sklearn syntax:
```suggestion
    def fit(self,X,y):
```